src/textarea/textarea.tsx

```diff
@@ -9,25 +10,120 @@ interface Props {
+    rows: number;
+    value: string;
+    limit: number;
+    stroke: string;
+    strokeDasharray: string;
+    outputText: string;
+    radius: number;
+}
+
+enum TextareaConfig {
+    minRows = 3,
+    maxRows = 10,
+    characterLimit = 256,
+}
+
+class Textarea extends React.Component<Props, State> {
+    private readonly DEFAULT_ROW_COUNT = 3;
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            rows: this.props.rows ? this.props.rows : this.DEFAULT_ROW_COUNT,
+            value: '',
+            limit: 255 - this.props.value.length,
+            stroke: '',
+            strokeDasharray: '',
+            outputText: '',
+            radius: 11,
+        };
+        this.onChange = this.onChange.bind(this);
+    }
+
+    public render() {
+        return (
+            <div className="textarea-component">
+                <textarea
+                    id={this.props.id}
+                    className={`form-control text-design ${this.state.stroke} ${
+                        this.props.enableCharacterLimit ? 'character-limited' : ''
+                    } ${this.props.className || ''}`}
+                    rows={this.state.rows}
+                    onBlur={this.props.onBlur}
+                    onChange={this.onChange}
+                    placeholder={this.props.placeholder}
+                    data-qa={`textarea--element-${this.props.id}`}
+                    value={this.props.value}
+                    disabled={this.props.disabled}
+                />
+                {this.props.enableCharacterLimit && (
+                    <CharacterLimitChecker
+                        radius={this.state.radius}
+                        stroke={this.state.stroke}
+                        strokeDasharray={this.state.strokeDasharray}
+                        limit={this.state.limit}
+                    />
+                )}
+            </div>
+        );
+    }
+
+    private onChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
+        this.props.onChange(event.target.value);
+        const currentRows = this.getCurrentRows(event, TextareaConfig.minRows, TextareaConfig.maxRows);
+        this.setState({
+            value: event.target.value,
+            rows: Math.min(currentRows, TextareaConfig.maxRows),
+            limit: this.getLimit(event),
+            stroke: this.isLimitExceeded(event) ? 'error-text' : 'normal-text',
+            strokeDasharray: `${this.getPercentage(event)} 999`,
+        });
+    }
+
+    private getCurrentRows(event: React.ChangeEvent<HTMLTextAreaElement>, minRows: number, maxRows: number) {
+        if (!this.props.enableCharacterLimit) {
+            return this.state.rows;
+        }
+        const textareaLineHeight = 14;
+
+        const previousRows = event.target.rows;
+        event.target.rows = minRows;
+
+        const currentRows = Math.floor(event.currentTarget.scrollHeight / textareaLineHeight);
+
+        if (currentRows === previousRows) {
```

Comment:
Can't you just do nothing in this case?

Reply:
the line-height style has to match with textareaLineHeight value so the text can be calculated based on it and this if condition is necessary  if (currentRows === previousRows)  to increase the row size